src/main/java/io/github/hapjava/services/impl/AbstractServiceImpl.java

```diff
@@ -28,6 +29,11 @@ public String getType() {
     return type;
   }
 
+  @Override
+  public List<Service> getLinkedServices() {
+    return new ArrayList<>();
+  }
+
```

Comment:
please make the linkedService list persistent by adding following
```suggestion

  private final List<Service> linkedServices = new LinkedList<>();
  @Override
  public void addLinkedService(Service s) { linkedServices.add(s);}

  @Override
  public List<Service> getLinkedServices() { return Collections.unmodifiableList(linkedServices); }
```

Reply:
Done